### DISCUSSION TITLE
Question: should switch_to_window visibly change focus on tabs when run headful?

### DISCUSSION THREAD

**Discussion by vdbijl** [rubycdp/cuprite#287] [Announcements]
I am debugging a problem that we have when testing with 2 tabs open. The test switches between tabs, but sometimes has a hard time finding elements on the screen.
When running the test headful I see that the switch_to_window does not visibly give the focus to the tab that is being switched to. Is that intentional or is that a bug?
Could the lack of focus be related to Cuprite/Capybara/Ferrum taking a long time finding elements on the non-focused page?

**Comment by route** [comment]
You should use `activate` https://github.com/rubycdp/ferrum/commit/600d63d0a2b1fd7787f69038ffeb9eabcc6c2f49
when switching tab. It should do the trick. Do you have a repro test?
